docs/design/transaction_flow.md

```diff
@@ -0,0 +1,67 @@
+    * Then call the `rooch_framework::transaction_validator::validate` method to verify the transaction's `Authenticator`. The `Authenticator` represents the `sender`'s proof of identity, usually a signature. The `scheme` field in the `Authenticator` is used to indicate the type of the `Authenticator`. Currently, two built-in `Authenticators`, `ED25519` and Ethereum's `ECDSA`, are supported, and more signature types will be supported in the future, as well as allowing developers to customize `Authenticator`. This is also part of AccountAbstraction. In addition, the `sequence_number`(equivalent to `nonce` in Ethereum) of the account will also be validated in `validate`.
+    * After validated by the contract, Executor constructs `TxContext` based on the transaction, uniformly converts the multi-chain transaction into `MoveOSTransaction`, and returns it.
+3. After receiving the return value of `validate_tx` from Executor, RPC Service decide whether the transaction has passed verification. If not, an error is returned directly to the user. Otherwise, the transaction is sent to Sequencer for `sequence_tx`. Because `validate_tx` is a read-only method and does not modify the state, returning an error at this step has no side effects.
+4. Sequencer adds the transaction to the Accumulator, obtains the `tx_order` of the transaction, and constructs `TransactionSequenceInfo`. `TransactionSequenceInfo` contains the signature of the `tx_order` assigned to the transaction by Sequencer and `tx_accumulator_root`. Sequencer periodically submits the transactions to DA in batches.
+5. After RPC Service receives the return value of `sequence_tx` from Sequencer, it sends the transaction to Executor for `execute_tx`.
+6. In Executor's `execute_tx`, MoveOS executes the transaction directly.
+    * First, MoveOS executes the `rooch_framework::transaction_validator::pre_execute` method for preprocessing the transaction. In the preprocessing, an account is automatically created and multi-chain addresses are mapped to Move addresses. In the future, Gas-related requirements in AccountAbstraction will also be implemented in the preprocessing, such as Gas exchange and Gas payment agents.
+    * Then, MoveOS calls the user-defined methods to execute the transaction.
+    * Finally, MoveOS executes the `rooch_framework::transaction_validator::post_execute` method for post-processing the transaction. In the post-processing, the account's `sequence_number` is updated automatically and Gas fees are deducted.
+    * During the execution process, `pre_execute`, `execute`, and `post_execute` share a `TxContext`, which can be used to pass data between them.
+    * Note that if the user-defined method fails during execution, MoveOS will automatically roll back the state, but `pre_execute` and `post_execute` will still be executed, and Gas fees will only be charged for the actual consumption of the user's execution.
```

Comment:
if the user-defined method fails during execution, the `pre_execute` and `post_execute` will still be executed, does `sequence_number`also  updated ?

Reply:
yes